src/krc_pb_client.erl

```diff
@@ -37,117 +37,198 @@
-  end.
-
-handle_stream(ReqId) ->
-  case wait_for_list(ReqId) of
-    {ok, Res}        ->  {ok, [krc_obj:decode(K) || K <- Res]};
-    {error, _} = Err -> Err
-  end.
+putwo(Pid, Obj, Options, Timeout) -> put(Pid, Obj, Options, Timeout).
+put(Pid, Obj, Options, Timeout)   ->
+  obj_exec(Pid, put, Obj, Options, Timeout).
 
 get_index(Pid, Bucket, Index, Key, Timeout) ->
   {Idx, IdxKey} = krc_obj:encode_index({Index, Key}),
-  case
-    riakc_pb_socket:get_index_eq(Pid,
-                                 krc_obj:encode(Bucket),
-                                 Idx,
-                                 IdxKey,
-                                 [{timeout, Timeout}, {stream, true}])
-  of
-    {ok, ReqId}      -> handle_stream(ReqId);
-    {error, _} = Err -> Err
-  end.
+  ExtraMetadata = #{index => Index, idx => Idx, idxkey => IdxKey},
+  StartMetadata = metadata(Pid, get_index_eq, Bucket, Key, [], Timeout, ExtraMetadata),
+  telemetry:span(
+    ?TELEMETRY_EVENT,
+    StartMetadata,
+    fun() ->
+        Resp = riakc_pb_socket:get_index_eq(Pid,
+                                            krc_obj:encode(Bucket),
+                                            Idx,
+                                            IdxKey,
+                                            [{timeout, Timeout}, {stream, true}]),
+        handle_resp(Resp, StartMetadata, fun handle_stream/1)
+    end
+   ).
 
 get_index(Pid, Bucket, Index, Lower, Upper, Timeout) ->
   {Idx, LowerKey} = krc_obj:encode_index({Index, Lower}),
   {Idx, UpperKey} = krc_obj:encode_index({Index, Upper}),
-  case
-    riakc_pb_socket:get_index_range(Pid,
-                                    krc_obj:encode(Bucket),
-                                    Idx,
-                                    LowerKey,
-                                    UpperKey,
-                                    [{timeout, Timeout}, {stream, true}])
-  of
-    {ok, ReqId}        ->  handle_stream(ReqId);
-    {error, _} = Err -> Err
-  end.
+  ExtraMetadata = #{index => Index, idx => Idx, lower_key => LowerKey, upper_key => UpperKey},
+  StartMetadata = metadata(Pid, get_index_range, Bucket, undefined, [], Timeout, ExtraMetadata),
+  telemetry:span(
+    ?TELEMETRY_EVENT,
+    StartMetadata,
+    fun() ->
+        Resp =
+          riakc_pb_socket:get_index_range(Pid,
+                                          krc_obj:encode(Bucket),
+                                          Idx,
+                                          LowerKey,
+                                          UpperKey,
+                                          [{timeout, Timeout}, {stream, true}]),
+        handle_resp(Resp, StartMetadata, fun handle_stream/1)
+    end
+   ).
 
 list_keys(Pid, Bucket, Timeout) ->
-    riakc_pb_socket:list_keys(Pid, krc_obj:encode(Bucket), Timeout).
+  StartMetadata =
+    #{request =>
+        #{pid => Pid, method => list_keys, bucket => Bucket, timeout => Timeout}
+     },
```

Comment:
Why not use the `metadata` functions with `undefined` values like you did on line 87 here too? Or make a new (metadata function) for the functions `list_keys`, `get_bucket` and `set_bucket` (props is the difference I can see here) ? They look similar to each other.